### PR TITLE
add documentation for the Posix shell

### PR DESCRIPTION
**Pull request by bkueng**
Follows from https://github.com/PX4/Firmware/pull/10173.

Points to discuss:
- Should we split `en/advanced/system_startup.md` into NuttX- & Posix-specific sections/pages?
- Should we move `en/advanced/system_startup.md` out of Advanced to e.g. Concepts?

### CODE REVIEW

File: en/advanced/system_startup.md [hunk 1]
```diff
@@ -1,10 +1,39 @@
 # System Startup
 
-The PX4 boot is controlled by shell scripts in the [ROMFS/px4fmu_common/init.d](https://github.com/PX4/Firmware/tree/master/ROMFS/px4fmu_common/init.d) folder.
+The PX4 startup is controlled by shell scripts.
+On NuttX they reside in the [ROMFS/px4fmu_common/init.d](https://github.com/PX4/Firmware/tree/master/ROMFS/px4fmu_common/init.d) folder.
+A portion of these is also used on Posix and the ones that are only used on Posix are located in [ROMFS/px4fmu_common/init.d-posix](https://github.com/PX4/Firmware/tree/master/ROMFS/px4fmu_common/init.d-posix).
```
Comment (hamishwillee) [inline en/advanced/system_startup.md]: Perhaps change to:

> On NuttX they reside in the [ROMFS/px4fmu_common/init.d](https://github.com/PX4/Firmware/tree/master/ROMFS/px4fmu_common/init.d) folder (some of these are also used on Posix).  The ones that are only used on Posix are located in [ROMFS/px4fmu_common/init.d-posix].

Should we refer to Posix as Posix/SITL to make it clear what we mean?

File: en/advanced/system_startup.md [hunk 2]
```diff
@@ -1,10 +1,39 @@
...
 
 All files starting with a number and underscore (e.g. `10000_airplane`) are canned airframe configurations. They are exported at build-time into an `airframes.xml` file which is parsed by [QGroundControl](http://qgroundcontrol.com) for the airframe selection UI. Adding a new configuration is covered [here](../airframes/adding_a_new_frame.md).
 
-The remaining files are part of the general startup logic, and the first executed file is the [rcS](https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/rcS) script, which calls all other scripts.
+The remaining files are part of the general startup logic, and the first executed file is the [init.d/rcS](https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/rcS) (or [init.d-posix/rcS](https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d-posix/rcS) on Posix) script, which calls all other scripts.
```
Comment (hamishwillee) [inline en/advanced/system_startup.md]: The remaining files are part of the general startup logic, and the first -> The remaining files are part of the general startup logic. The first

File: en/advanced/system_startup.md [hunk 3]
```diff
@@ -1,10 +1,38 @@
 # System Startup
 
-The PX4 boot is controlled by shell scripts in the [ROMFS/px4fmu_common/init.d](https://github.com/PX4/Firmware/tree/master/ROMFS/px4fmu_common/init.d) folder.
+The PX4 startup is controlled by shell scripts.
+On NuttX they reside in the [ROMFS/px4fmu_common/init.d](https://github.com/PX4/Firmware/tree/master/ROMFS/px4fmu_common/init.d) folder - some of these are also used on Posix (Linux/MacOS). The scripts that are only used on Posix are located in [ROMFS/px4fmu_common/init.d-posix](https://github.com/PX4/Firmware/tree/master/ROMFS/px4fmu_common/init.d-posix).
```
Comment (hamishwillee) [inline en/advanced/system_startup.md]: I changed to "used on Posix **(Linux/MacOS)**" - ie to add the meaning of posix first time it is used. Slightly leaning towards referring this as "Simulation"

**Comment by hamishwillee** [comment]
@bkueng Great job - certainly this is understandable. I have done a minor subedit.

> Should we split en/advanced/system_startup.md into NuttX- & Posix-specific sections/pages?

My gut feeling is yes to separate sections (not pages).

> Should we move en/advanced/system_startup.md out of Advanced to e.g. Concepts?

Definitely. In first instance this could be done just by moving the doc in the summary.

Question, is this a pure "new system developer" topic? ie do we need to think about moving this to the developer guide?

File: en/advanced/system_startup.md (hunk 3 above)
Comment (bkueng) [reply en/advanced/system_startup.md]: This is good. It's not only simulation, as in future our Linux-based boards (such as RPi) should use that as well.

**Review by bkueng — COMMENTED** [review]
> I have done a minor subedit.

Thanks.
I'll split into sections then.

> Question, is this a pure "new system developer" topic? ie do we need to think about moving this to the developer guide?

This is the dev-guide :)

**Comment by bkueng** [comment]
I did the splitting and moved it to Concepts.
I also changed the NuttX boot: driver failures will not lead to a aborted boot anymore. This was changed a while ago.

**Comment by hamishwillee** [comment]
Looks great thanks Beat. 

FYI only I did a minor additional subedit, moved the doc to concepts and created a redirect (decided it is worth it) and copied doc into Korean/Chinese tree (a bit pointless since we're moving to slightly different approach, but hey!).

**Pull request merged**